### ISSUE TITLE
Add date of last successfull rss parsing

### ISSUE THREAD

**Issue by tdziurko** [jvm-bloggers/jvm-bloggers#582]
Some RSS feeds are incorrect (not a valid RSS), some simply stops working over time (domain expiry, etc.) so we should be able to find such dead/invalid entries and investigate them to notify the author or disable them from the system.

So it would be nice to have a field with information when it was the last time we were able to parse given RSS feed successfully.

**Comment by tdziurko** [comment]
@mirogordon is on it :)

**Comment by malochak** [comment]
Is it still open?

**Comment by tdziurko** [comment]
Yes, you can work on it :)

**Comment by malochak** [comment]
@tdziurko I went through contributing, code-style importing, and initialisation related stuff. 

I've started looking at the code, I think I found a place where the update should be done. However, I'm not sure if the solution is correct and some questions also appeared in my head :)

1. I assume that `BlogPostsFetcher#refreshPosts` is the entry point. It might be called manually from the admin panel and it is also triggered by the scheduler in given time intervals. 
2. `SyndFeedProducer#createFor` is the place where the RSS feed is tried to be fetched by all available fetchers (HTTP, HTTPS, WGET...), then the XML is parsed. I guess that more than one fetcher can finish job successfully.
3. So, in `SyndFeedProducer#createFor` I could do something like this (forgive me for any mistakes - it's my first contact with `vavr`):
```java
public void checkForNewEntries(Blog person) {
  log.info("Checking for new entries at {} by {}.", person.getRss(), person.getAuthor());

  syndFeedFactory.createFor(person.getRss())
      .toJavaList()
      .stream()
      .peek(feed -> {
        log.info("For author {} found {} entries in rss feed", person.getAuthor(),
            feed.getEntries().size());
        feed.getEntries().forEach(post -> {
          blogPostService.addOrUpdate(new RssEntryWithAuthor(person, post));
        });
      }).findAny().ifPresent(feed -> {
        log.info("For author {} updated date last fetched to {}", person.getAuthor(),
            LocalDateTime.now());
        blogRepository.updateDateLastFetched(LocalDateTime.now(), person.getId());
      });
}
```
4. We also need a new field that will keep information about last successful fetch
5. Because of 4th point we need also DB migration - not sure how to prepare such, is `src/main/resources/db/changelog` the place that I should look at?
6. unit tests obviously :)

**Comment by tdziurko** [comment]
Hello @malochak
1. Assumption correct :)
2. I don't think it could be executed more than once. Parsers are executed in order and next one is called if the previous ones haven't succeeded. If I am clearly mistaken, please let me know :) 
3. More or less something like this, but I think if your point 2 is incorrect, it will look a little bit different.
4 and 5: Yes, new field on Blog entity I guess and changelog dir is the please you are looking for.
6. Please, do unit test :) Can be Spock, can be new JUnit, up to you.

**Comment by malochak** [comment]
@tdziurko You're right about the parsers part. There're tests that confirms that. So it means that I don't need to modify so much `SingleRssChecker`, just need to put the update call at the end of processing.

**Comment by malochak** [comment]
@tdziurko I cannot push my branch with fix 
```
ERROR: Permission to jvm-bloggers/jvm-bloggers.git denied to malochak.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by airborn** [comment]
Please fork the repo, push to your own fork and create a pull request from there

**Comment by malochak** [comment]
> Please fork the repo, push to your own fork and create a pull request from there

Thank's, you're right. I've just cloned the repo. Never contributed to open source project, thought I'm just allowed to push a branch and then create PR 🤦‍♂️
